File: javascript/3301.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split(' ');
var h = parseInt(lines.shift()); //huguinho
var z = parseInt(lines.shift()); //zezinho
var l = parseInt(lines.shift()); //luisinho

if((h > z && h < l) || (h < z && h > l)) {
     console.log("huguinho")
} else if((z > h && z < l) || (z < h && z > l)) {
     console.log("zezinho")
} else {
     console.log("luisinho")
}

File: python/3301.py
i=input().split(" ")
h=int(i[0])
z=int(i[1])
l=int(i[2])

if (h > z and h < l) or (h < z and h > l):
    print("huguinho")
elif (z > h and z < l) or (z < h and z > l):
    print("zezinho")
else:
    print("luisinho")